llrt_modules/src/modules/zlib.rs

```diff
@@ -0,0 +1,293 @@
+    }
+
+    let mut dst: Vec<u8> = Vec::with_capacity(src.len());
+
+    let _ = match command {
+        ZstdCommand::Compress => ZstdEncoder::new(&src[..], level)?.read_to_end(&mut dst)?,
+        ZstdCommand::Decompress => ZstdDecoder::new(&src[..])?.read_to_end(&mut dst)?,
+    };
+
+    Buffer(dst).into_js(&ctx)
+}
+
+define_cb_function!(zstd_comp, zstd_converter, ZstdCommand::Compress);
+define_sync_function!(zstd_comp_sync, zstd_converter, ZstdCommand::Compress);
+
+define_cb_function!(zstd_decomp, zstd_converter, ZstdCommand::Decompress);
+define_sync_function!(zstd_decomp_sync, zstd_converter, ZstdCommand::Decompress);
+
+pub struct ZlibModule;
+
+impl ModuleDef for ZlibModule {
+    fn declare(declare: &Declarations) -> Result<()> {
+        declare.declare("deflate")?;
+        declare.declare("deflateSync")?;
+
+        declare.declare("deflateRaw")?;
+        declare.declare("deflateRawSync")?;
+
+        declare.declare("gzip")?;
+        declare.declare("gzipSync")?;
+
+        declare.declare("inflate")?;
+        declare.declare("inflateSync")?;
+
+        declare.declare("inflateRaw")?;
+        declare.declare("inflateRawSync")?;
+
+        declare.declare("gunzip")?;
+        declare.declare("gunzipSync")?;
+
+        declare.declare("brotliCompress")?;
+        declare.declare("brotliCompressSync")?;
+
+        declare.declare("brotliDecompress")?;
+        declare.declare("brotliDecompressSync")?;
+
+        declare.declare("zstandardCompress")?;
+        declare.declare("zstandardCompressSync")?;
+
+        declare.declare("zstandardDecompress")?;
+        declare.declare("zstandardDecompressSync")?;
+
+        declare.declare("default")?;
+        Ok(())
+    }
+
+    fn evaluate<'js>(ctx: &Ctx<'js>, exports: &Exports<'js>) -> Result<()> {
+        export_default(ctx, exports, |default| {
+            default.set("deflate", Func::from(deflate))?;
+            default.set("deflateSync", Func::from(deflate_sync))?;
+
+            default.set("deflateRaw", Func::from(deflate_raw))?;
+            default.set("deflateRawSync", Func::from(deflate_raw_sync))?;
+
+            default.set("gzip", Func::from(gzip))?;
+            default.set("gzipSync", Func::from(gzip_sync))?;
+
+            default.set("inflate", Func::from(inflate))?;
+            default.set("inflateSync", Func::from(inflate_sync))?;
+
+            default.set("inflateRaw", Func::from(inflate_raw))?;
+            default.set("inflateRawSync", Func::from(inflate_raw_sync))?;
+
+            default.set("gunzip", Func::from(gunzip))?;
+            default.set("gunzipSync", Func::from(gunzip_sync))?;
+
+            default.set("brotliCompress", Func::from(br_comp))?;
+            default.set("brotliCompressSync", Func::from(br_comp_sync))?;
+
+            default.set("brotliDecompress", Func::from(br_decomp))?;
+            default.set("brotliDecompressSync", Func::from(br_decomp_sync))?;
+
+            default.set("zstandardCompress", Func::from(zstd_comp))?;
```

Comment:
Idea @richarddavison we might want to add a comment or macro to tag APIs to know if they respect a certain spec or existing engine. That way we could auto generate a report or something. Like this method is good but its not a node thing (yet). Maybe a rustdoc tag like `/// @engine ` or something.

Reply:
Hi @Sytten , Sorry, I did not know what exactly to do about this point...
Should I put the annotation in the rust source? Or should I put the annotations in the types?

As for Zstandard, it is not so difficult to implement, and once Node.js supports it, there is a way to support it in LLRT as well.

Reply:
Do nothing for now :)
This is just an idea so we can track better from which spec each API comes from.